.github/workflows/preview.yml

```diff
@@ -0,0 +1,99 @@
+name: Preview PR
+
+on:
+  workflow_run:
+    workflows: ["Build and publish"]
+    types: 
+      - completed
+
+jobs:
+# This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-20.04
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+      
+      # Extracts PR from Head SHA
+      - name: Find PR
+        if: ${{ github.event.workflow_run.event == 'pull_request' }}
+        run: |
+          PR=$(curl https://api.github.com/search/issues?q=${{ github.event.workflow_run.head_sha }} |
+          grep -Po "(?<=${{ github.event.workflow_run.repository.full_name }}\/pulls\/)\d*" | head -1)
+          echo "PR=$PR" >> $GITHUB_ENV
+      
+      # Replaces base specs with head specs
+      - name: Pull PR
+        if: ${{ github.event.workflow_run.event == 'pull_request' && env.PR }}
+        run: |
+          rm spec/index.bs
+          rm wgsl/index.bs
+          wget -O spec/index.bs https://raw.githubusercontent.com/${{ github.event.workflow_run.head_repository.full_name }}/${{ github.event.workflow_run.head_sha }}/spec/index.bs
+          wget -O wgsl/index.bs https://raw.githubusercontent.com/${{ github.event.workflow_run.head_repository.full_name }}/${{ github.event.workflow_run.head_sha }}/wgsl/index.bs
+      
+      # Adds Firebase config files to directory
+      - name: Init Firebase
+        if: ${{ github.event.workflow_run.event == 'pull_request' && env.PR }}
+        run: |
+          cat << EOF >> .firebaserc
+          {
+            "projects": {
+              "default": "gpuweb-prs"
+            }
+          }
+          EOF
+          cat << EOF >> firebase.json
+          {
+            "hosting": {
+              "public": "out",
+              "ignore": [
+                "firebase.json",
+                "**/.*",
+                "**/node_modules/**"
+              ]
+            }
+          }
+          EOF
+
+      # Builds Bikeshed specs
+      - name: Build Bikeshed
+        if: ${{ github.event.workflow_run.event == 'pull_request' && env.PR }}
+        run: |
+          git clone --depth 1 https://github.com/tabatkins/bikeshed.git
+          pip3 install --user --editable ./bikeshed
+          export PATH="$(python3 -m site --user-base)/bin:${PATH}"
```

Comment:
Rather than cloning bikeshed, you can simply install directly from PyPI via `pip3 install bikeshed` (should replace the above 3 lines). (You're adding a dependency on pypi.org being online.)

Reply:
Also, to be clear, it's perfectly fine to curl to the bikeshed API here rather than installing locally.

Reply:
Thanks!